packages/pluggableWidgets/image-viewer-web/src/components/ImageViewer.tsx

```diff
@@ -0,0 +1,80 @@
+import { createElement, CSSProperties, ReactElement } from "react";
+import classNames from "classnames";
+import { HeightUnitEnum, WidthUnitEnum } from "../../typings/ImageViewerProps";
+
+import "../ui/ImageViewer.scss";
+
+function getStyle(value: string | number, type: WidthUnitEnum | HeightUnitEnum): number | string {
+    console.log({ value, type });
+    // when type is auto default browser styles applies
+    if (type === "pixels") {
+        return value;
+    } else if (type === "percentage") {
+        return value + "%";
+    }
+
+    return "";
+}
+export interface ImageViewerWrapperProps {
+    className?: string;
+    responsive: boolean;
+    hasImage: boolean;
+    children: ReactElement<ImageViewerGlyphicon | ImageViewerImage>;
+}
+
+interface ImageViewerContent {
+    style?: CSSProperties;
+}
+
+function Wrapper(props: ImageViewerWrapperProps): ReactElement {
+    return (
+        <div
+            className={classNames(
+                "mx-image-viewer",
+                { "mx-image-viewer-responsive": props.responsive },
+                props.className,
+                { hidden: !props.hasImage }
+            )}
+        >
+            {props.children}
+        </div>
+    );
+}
+
+export interface ImageViewerGlyphicon extends ImageViewerContent {
+    icon: string | undefined;
+    size: number;
+}
+
+function Glyphicon(props: ImageViewerGlyphicon): ReactElement {
+    return (
+        <span className={classNames("glyphicon", props.icon)} style={{ ...props.style, fontSize: `${props.size}px` }} />
+    );
+}
+
+export interface ImageViewerImage extends ImageViewerContent {
+    image: string | undefined;
+    height: number;
+    heightUnit: HeightUnitEnum;
+    width: number;
+    widthUnit: WidthUnitEnum;
+}
+
+function Image(props: ImageViewerImage): ReactElement {
+    return (
+        <img
+            src={props.image}
+            style={{
+                ...props.style,
+                height: getStyle(props.height, props.heightUnit),
+                width: getStyle(props.width, props.widthUnit)
+            }}
+        />
+    );
+}
+
+export const ImageViewer = {
```

Comment:
Nice way of doing it this way. Is there a specifc reason why you went for this approach?

Reply:
1. Wanted to try this form of composition out, so basically curiosity
2. Practically speaking, these components are created to be used together. So exporting them like this forces you to acknowledge and be aware of that fact. The components are specifically made for these use cases and it doesn't really make sense to use in other scenarios _the way they are now_.
3. The reason I'm creating multiple components is because otherwise I have to put multiple conditionals inside this component and add another enum prop for `imageType` for example. This is basically prop drilling and also adds a lot of additional props that are unnecessary. (Do you use the same prop for image url and glypicon classname? Or do you create separate props? In that case you create unreachable combinations of props. What if we add a different type of image in the feature?) This also creates a more clear separation of concerns between the container (`ImageViewer`) and the content (the other two components). Basically this is the composition pattern part.
4. Using the above pattern, it's also easier to control stuff like whether an image type should include a lightbox and to extend the type of content we support.